src/client/client_config.c

```diff
@@ -162,7 +163,12 @@ int client_config(client_config_coll_t *ccc, int global_config)
             ccc->profiles = g_array_new(FALSE, FALSE, sizeof(
                                             struct profile_config_s));
         
-        if (NULL == (tmp_str = getenv(LIXA_PROFILE_ENV_VAR))) {
+        if (profile) {
```

Comment:
Putting "if (profile)" in the first stage overrides LIXA_PROFILE environment variable: is this the desired behavior? Have you thought about looking at "profile" passed argument as a second choice and leaving precedence to the environment variable?
I could imagine this scenario:
- an Application Program uses the new extension to point a specific profile, wherever the profile string is coded (some configuration file of the application, hard coded, whatever...)
- under some circumstances operations people want to bypass standard configuration through LIXA_PROFILE environment variable without recompiling or touching the application configuration.
Does it make sense? Have you made some reasoning about it for a generic use of the API?
I'm pointing out this topic because some LIXA environment variables have precedence over the corresponding parameters in LIXA configuration files: I want to be sure your proposal is the best alternative in a real life scenario.

Reply:
I had really envisaged this as either-or, rather than an override. So yes, letting LIXA_PROFILE override the application-set profile might make more sense and does not affect our use case.

Reply:
client_config.c changed to make LIXA_PROFILE override lixa_tx_set_profile() added some additional tests.
I have also reverted the case0038.c changes - the tests still run.